### PR TITLE
Make IMixedRealityControllerVisualizer less implementation-specific

### PR DESCRIPTION
**Pull request by keveleigh**
Overview
---
Not all visualizers will be driven by a pose synchronizer. Implementations that want to be (like `MixedRealityControllerVisualizer`) are still able to implement both interfaces.

### CODE REVIEW

**Comment by davidkline-ms** [comment]
> Needs to have a design discussion

@keveleigh, can you describe a concrete example of how this is helpful?

File: Assets/MixedRealityToolkit/_Core/Interfaces/Devices/IMixedRealityControllerVisualizer.cs [hunk 1]
```diff
@@ -15,6 +15,11 @@ public interface IMixedRealityControllerVisualizer : IMixedRealityControllerPose
         /// </remarks>
         GameObject GameObjectProxy { get; }
 
+        /// <summary>
+        /// The current controller reference.
+        /// </summary>
+        IMixedRealityController Controller { get; set; }
```
Comment (SimonDarksideJ) [inline Assets/MixedRealityToolkit/_Core/Interfaces/Devices/IMixedRealityControllerVisualizer.cs]: I see no reason to attach a physical controller in to the abstract implementation for visualization.

The events already give access to the source controller / data.

And the current visualizer has dependencies on the SourcePoseSynchroniser

**Comment by StephenHodgson** [comment]
Also let's focus on bug fixes for stabilization.

This isn't something we **need** to release on beta.

File: Assets/MixedRealityToolkit/_Core/Interfaces/Devices/IMixedRealityControllerVisualizer.cs (hunk 1 above)
Comment (keveleigh) [reply Assets/MixedRealityToolkit/_Core/Interfaces/Devices/IMixedRealityControllerVisualizer.cs]: This is needed for a line in `BaseController`, which only references this interface: https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/mrtk_development/Assets/MixedRealityToolkit/_Core/Devices/BaseController.cs#L197

**Comment by SimonDarksideJ** [comment]
Re-targetted back to development for further discussion. Not a beta candidate

**Comment by SimonDarksideJ** [comment]
@keveleigh do you think you could close this PR and start up a new task to discuss the requirements for this change, reasons why the current implementation is not sufficient and any other alternatives you've considered.
That is supposed to be the process for introducing change in to the framework as I understand it.

**Comment by StephenHodgson** [comment]
Agreed. Also, visualization is going to need to be synced with the controller pose. I'm not sure this is a change we should be making.

**Pull request closed without merging**